### DISCUSSION TITLE
Login hooks?

### DISCUSSION THREAD

**Discussion by howesteve** [supabase/supabase#9337] [General]
Hello.

Is it possible to implement login(/logout) hooks? I need to run a database function or server side code after a user logs in. 
That would allow me to sent a security email whenever users log in, check it's payment status, etc.

Thanks

**Comment by GaryAustin1** [comment]
You would need to attempt this with trigger functions on the auth users table being updated.  Login should be easy as the there is a last sign in date column you could check to see if it is being updated and then do what you want to do in the trigger function.

Not sure how to detect signout. 

https://github.com/supabase/supabase/discussions/8814

**Reply by howesteve** [reply]
Ok, that's a good idea. I just realized now ```auth.users``` has a ```last_sign_in_at``` field and a trigger would be suitable. However for signout there is still no way to do it. That could be an enhacement.
Thank you!
